sqlalchemy_utils/functions/database.py

```diff
@@ -420,6 +421,39 @@ def is_auto_assigned_date_column(column):
     )
 
 
+def _set_url_database(url: sa.engine.url.URL, database):
+    if hasattr(sa.engine, 'URL'):
+        ret = sa.engine.URL.create(
+            drivername=url.drivername,
+            username=url.username,
+            password=url.password,
+            host=url.host,
+            port=url.port,
+            database=database,
+            query=url.query
+        )
+    else:  # SQLAlchemy <1.4
+        url.database = database
+        ret = url
+    assert ret.database == database, ret
+    return ret
```

Comment:
@nsoranzo Hi, thanks for working on this.

Tried utilizing your PR in my project, but seems this commit (
dd720c0) is causing a regression (or very similar bug) of #462 except for with the `create_database` utility. 

```python
@pytest.fixture(scope="session", autouse=True)
def create_test_database():
    root = Path(__file__).parent.parent.parent
    alembic_cfg = root / 'alembic.ini'
    url = str(settings.TEST_DATABASE_URI)
    assert alembic_cfg.exists(), f"Failed to locate alembic config: {alembic_cfg}"
    assert not database_exists(url), f"Testing database ({url}) already exists!"
    create_database(url)
    config = Config(str(alembic_cfg))
    config.set_main_option('sqlalchemy.url', url)
    command.upgrade(config, 'head')
    yield
    drop_database(url)
```


Traceback in spoiler below.

<details>
  <summary>Traceback</summary>

```sh

../.direnv/python-3.8.8/lib/python3.8/site-packages/psycopg2/__init__.py:127: OperationalError

The above exception was the direct cause of the following exception:

    @pytest.fixture(scope="session", autouse=True)
    def create_test_database():
        root = Path(__file__).parent.parent.parent
        alembic_cfg = root / 'alembic.ini'
        url = str(settings.TEST_DATABASE_URI)
        assert alembic_cfg.exists(), f"Failed to locate alembic config: {alembic_cfg}"
        assert not database_exists(url), f"Testing database ({url}) already exists!"
>       create_database(url)

myproject/tests/conftest.py:46:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy_utils/functions/database.py:570: in create_database
    with engine.connect() as connection:
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/base.py:3067: in connect
    return self._connection_cls(self, close_with_result=close_with_result)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/base.py:91: in __init__
    else engine.raw_connection()
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/base.py:3146: in raw_connection
    return self._wrap_pool_connect(self.pool.connect, _connection)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/base.py:3116: in _wrap_pool_connect
    Connection._handle_dbapi_exception_noconnection(
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/base.py:2003: in _handle_dbapi_exception_noconnection
    util.raise_(
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/util/compat.py:198: in raise_
    raise exception
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/base.py:3113: in _wrap_pool_connect
    return fn()
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:301: in connect
    return _ConnectionFairy._checkout(self)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:755: in _checkout
    fairy = _ConnectionRecord.checkout(pool)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:419: in checkout
    rec = pool._do_get()
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/impl.py:145: in _do_get
    self._dec_overflow()
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py:70: in __exit__
    compat.raise_(
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/util/compat.py:198: in raise_
    raise exception
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/impl.py:142: in _do_get
    return self._create_connection()
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:247: in _create_connection
    return _ConnectionRecord(self)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:362: in __init__
    self.__connect(first_connect_check=True)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:605: in __connect
    pool.logger.debug("Error on connect(): %s", e)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py:70: in __exit__
    compat.raise_(
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/util/compat.py:198: in raise_
    raise exception
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/pool/base.py:599: in __connect
    connection = pool._invoke_creator(self)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/create.py:578: in connect
    return dialect.connect(*cargs, **cparams)
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy/engine/default.py:548: in connect
    return self.dbapi.connect(*cargs, **cparams)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

dsn = 'host=localhost user=myuser password=mypass', connection_factory = None, cursor_factory = None, kwargs = {'host': 'localhost', 'password': 'mypass', 'user': 'myuser'}, kwasync = {}

    def connect(dsn=None, connection_factory=None, cursor_factory=None, **kwargs):

    ## REMOVED FOR BREVITY ##

        dsn = _ext.make_dsn(dsn, **kwargs)
>       conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
E       sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) FATAL:  database "myuser" does not exist
E
E       (Background on this error at: http://sqlalche.me/e/14/e3q8)

../.direnv/python-3.8.8/lib/python3.8/site-packages/psycopg2/__init__.py:127: OperationalError

```
</details>


Inspected with pdb, and looks like this check may not be valid on the latest `1.4.3`

```sh
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /myproject/.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy_utils/functions/database.py(426)_set_url_database()
-> if hasattr(sa.engine, 'URL'):
(Pdb) hasattr(sa.engine, 'URL')
True
(Pdb) sa.__version__
'1.4.3'
(Pdb) url
postgresql://myuser:***@localhost/mydb_test
(Pdb)

```

Reply:
Actually... I just tried this same code on ebf0c32 and received the same error.

:thinking:

Reply:
Managed to solve the problem. See my PR on your fork [here](https://github.com/nsoranzo/sqlalchemy-utils/pull/1).